javalin/src/main/java/io/javalin/http/JavalinServlet.kt

```diff
@@ -115,9 +119,14 @@ class JavalinServlet(val config: JavalinConfig) : HttpServlet() {
     private fun hasGetHandlerMapped(requestUri: String) = matcher.findEntries(HandlerType.GET, requestUri).isNotEmpty()
 
     fun addHandler(handlerType: HandlerType, path: String, handler: Handler, roles: Set<Role>) {
-        val shouldWrap = handlerType.isHttpMethod() && !roles.contains(CoreRoles.NO_WRAP)
+        val shouldWrap = handlerType.isHttpMethod()
         val protectedHandler = if (shouldWrap) Handler { ctx -> config.inner.accessManager.manage(handler, ctx, roles) } else handler
         matcher.add(HandlerEntry(handlerType, path, protectedHandler, handler))
     }
 
+    private fun isCorsEnabled(config: JavalinConfig) = try {
+        config.getPlugin(CorsPlugin::class.java) != null
+    } catch (ignored: PluginNotFoundException) {
+        false
+    }
```

Comment:
I'm not sure I like testing for the plugin presence via a try/catch , but I couldn't find a simpler way to do so.

Reply:
I'm surprised `getPlugin` feels the need to throw an exception, but I guess I approved of that change at some point. You can do this:

```kotlin
    private fun isCorsEnabled(config: JavalinConfig) = 
            config.inner.plugins.get(CorsPlugin::class.java) != null
```